### PR TITLE
Adds missing __init__.py files [FC-0030]

### PR DESCRIPTION
**Pull request by pomegranited**
Adds missing `__init__.py` files, and fixes the resulting linting errors, plus some minor test fixes:

* ensures 100% test coverage for files under `tests/` ([see previous coverage report](https://github.com/openedx/openedx-learning/actions/runs/6398610152/job/17369107033#step:7:160))
* fixes PytestCollectionWarnings ([see previous warnings](https://github.com/openedx/openedx-learning/actions/runs/6398610152/job/17369107033#step:7:49))

This change shouldn't require any manual testing, just a code review.

Follow-up to https://github.com/openedx/openedx-learning/pull/87#discussion_r1344741735

Private-ref: [FAL-3477](https://tasks.opencraft.com/browse/FAL-3477)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Review by ChrisChV — APPROVED** [review]
@pomegranited Looks good to me, thanks for this work :+1:
- [X] I tested this: A ran the tests and checked the coverage and warnings
- [X] I read through the code and considered the security, stability and performance implications of the changes.
- [X] Includes documentation.

File: tests/openedx_tagging/core/tagging/test_views.py [hunk 1]
```diff
@@ -590,7 +590,7 @@ def test_object_tags_remaining_http_methods(
             response = self.client.post(url, {"test": "payload"}, format="json")
         elif http_method == "PATCH":
             response = self.client.patch(url, {"test": "payload"}, format="json")
-        elif http_method == "DELETE":
+        else:  # http_method == "DELETE"
```
Comment (bradenmacdonald) [inline tests/openedx_tagging/core/tagging/test_views.py:593]: This is the only change I'm not sure about... how do we know it's `DELETE` here and not `PUT` for example? Especially for something as important as deleting, I'd prefer to be explicit about matching the DELETE method before assume it's delete.

**Review by bradenmacdonald — CHANGES_REQUESTED** [review]
Thanks, very nice cleanup! Just one little thing that looks risky to me, and otherwise it's awesome and I'll approve and merge.

File: tests/openedx_tagging/core/tagging/test_views.py (hunk 1 above)
Comment (pomegranited) [reply tests/openedx_tagging/core/tagging/test_views.py]: This should make it better: [868057d](https://github.com/openedx/openedx-learning/pull/90/commits/868057dcba4471f89e62fe7afa4bb9798175d967)

**Review by bradenmacdonald — APPROVED** [review]
Thanks, very nice work! Please fix the conflict and merge at your convenience.

**Comment by openedx-webhooks** [comment]
@pomegranited 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-learning&entry.867055334=https://github.com/openedx/openedx-learning/pull/90&entry.1484655318=https://github.com/pomegranited&entry.752974735=2023-10-04+03:39&entry.1917517419=2023-10-09+00:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**